### PR TITLE
BinaryMask Brush-related Functions & Debugging Improvements

### PR DESCRIPTION
**Pull request by Penguin5**
Added BinaryMask functions: combineWithOffset, combineWithOffset, connectSpawnsWithRandomConsecutiveBrushUse, useBrushRepeatedlyForRandomConsecutiveExpansion and addFloatMaskValuesWithinRangeCenteredAtLocationWithSize
Added ConcurrentBinaryMask functions: useBrushRepeatedlyForRandomConsecutiveExpansion and connectSpawnsWithRandomConsecutiveBrushUse
Changed numerous functions' mask size error messages to be more detailed
Added automatically named visual debugging options
Added seed parameter for loadBrush function

### CODE REVIEW

File: src/java/map/FloatMask.java [hunk 1]
```diff
@@ -907,6 +907,10 @@ public String toHash() throws NoSuchAlgorithmException {
     }
 
 
+    public FloatMask startVisualDebugger() {
+        return startVisualDebugger(toString(), Util.getStackTraceParentClass());
+    }
+
```
Comment (Sheikah45) [inline src/java/map/FloatMask.java:913]: Why was this added the default toString is not that meaningful

File: src/java/map/ConcurrentFloatMask.java [hunk 2]
```diff
@@ -238,6 +238,11 @@ public void show() {
         this.floatMask.show();
     }
 
+    public ConcurrentFloatMask startVisualDebugger() {
+        this.floatMask.startVisualDebugger(Util.getStackTraceParentClass());
+        return this;
+    }
+
```
Comment (Sheikah45) [inline src/java/map/ConcurrentFloatMask.java:245]: Why was this added masks need a name in the debugger?

File: src/java/map/ConcurrentBinaryMask.java [hunk 3]
```diff
@@ -301,6 +315,11 @@ public void show() {
         this.binaryMask.show();
     }
 
+    public ConcurrentBinaryMask startVisualDebugger() {
+        this.binaryMask.startVisualDebugger(Util.getStackTraceParentClass());
+        return this;
+    }
+
```
Comment (Sheikah45) [inline src/java/map/ConcurrentBinaryMask.java:322]: same

File: src/java/map/FloatMask.java (hunk 1 above)
Comment (Penguin5) [reply src/java/map/FloatMask.java:913]: It has some relevant information and is easier/faster for me to use than having to manually write a name each time I want to visually debug something.

File: src/java/map/ConcurrentFloatMask.java (hunk 2 above)
Comment (Penguin5) [reply src/java/map/ConcurrentFloatMask.java:245]: it uses the toString() name from the FloatMask function I made...

File: src/java/map/ConcurrentBinaryMask.java (hunk 3 above)
Comment (Penguin5) [reply src/java/map/ConcurrentBinaryMask.java:322]: it uses the toString() name from the BinaryMask function I made...

File: src/java/map/BinaryMask.java [hunk 4]
```diff
@@ -168,6 +173,51 @@ public BinaryMask flipValues(float density, Symmetry symmetry) {
         return this;
     }
 
+    public BinaryMask addFloatMaskValuesWithinRangeCenteredAtLocationWithSize(FloatMask other, float minValueToConvert, float maxValueToConvert, Vector2f location, int size) {
+        if (size > getSize()) {
+            throw new IllegalArgumentException("Masks not the same size: other is " + other.getSize() + " and BinaryMask is " + getSize());
```
Comment (Sheikah45) [inline src/java/map/BinaryMask.java]: This is checking for a greater than condition so the error should reflect that

File: src/java/map/BinaryMask.java [hunk 5]
```diff
@@ -168,6 +173,51 @@ public BinaryMask flipValues(float density, Symmetry symmetry) {
...
+            y += random.nextBoolean() ? random.nextInt(maxDistanceBetweenBrushstrokeCenters + 1) : - random.nextInt(maxDistanceBetweenBrushstrokeCenters + 1);
+            combineWithOffset(brush, x, y, true);
+        }
+        VisualDebugger.visualizeMask(this);
+        return this;
+    }
+
+    public BinaryMask connectSpawnsWithRandomConsecutiveBrushUse(ArrayList<Spawn> spawns, int percentChanceToAttemptConnectionPerOddNumberedSpawn, String brushName, int size, int numberOfUsesBatchSize, float minIntensityForTrue, float maxIntensityForTrue, int maxDistanceBetweenBrushstrokeCenters) {
+        for (int z = 0; z < spawns.size() / 2; z+=2) {
+            if(percentChanceToAttemptConnectionPerOddNumberedSpawn > random.nextInt(100) - 1) {
```
Comment (Sheikah45) [inline src/java/map/BinaryMask.java]: percentChance should be a float and then use random.nextFloat for condition check

**Review by Sheikah45 — CHANGES_REQUESTED** [review]

File: src/java/map/BinaryMask.java (hunk 4 above)
Comment (Penguin5) [reply src/java/map/BinaryMask.java]: fixed

File: src/java/map/BinaryMask.java (hunk 5 above)
Comment (Penguin5) [reply src/java/map/BinaryMask.java]: changed

**Review by Sheikah45 — APPROVED** [review]
